### PR TITLE
Refactor `ScriptArguments`

### PR DESCRIPTION
**Pull request by qgallouedec**
# What does this PR do?

Script arguments class is defined

- Directly in scripts, like 
https://github.com/huggingface/trl/blob/70036bf87f1036ef43b92cc421f7a3049debb1ec/examples/scripts/orpo.py#L64-L69
- with `SFTScriptArguments`:
https://github.com/huggingface/trl/blob/70036bf87f1036ef43b92cc421f7a3049debb1ec/trl/commands/cli_utils.py#L82-L94
- with `DPOScriptArguments`:
https://github.com/huggingface/trl/blob/70036bf87f1036ef43b92cc421f7a3049debb1ec/trl/commands/cli_utils.py#L112-L129
- with `RewardScriptArguments`:
https://github.com/huggingface/trl/blob/70036bf87f1036ef43b92cc421f7a3049debb1ec/trl/commands/cli_utils.py#L97-L109

They are pretty much the same. I suggest:

1. Merging everything into one `ScriptArguments`
2. Deprecate `SFTScriptArguments`, `DPOScriptArguments` and `RewardScriptArguments` in favor of `ScriptArguments`.
3. Standardize the use of `ScriptArguments` to all examples scripts.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/trl/tree/main/docs).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: trl/utils.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    dataset_train_split (`str`, *optional*, defaults to `"train"`):
+        Dataset split to use for training.
+    dataset_test_split (`str`, *optional*, defaults to `"test"`):
+        Dataset split to use for evaluation.
+    config (`str` or `None`, *optional*, defaults to `None`):
+        Path to the optional config file.
+    gradient_checkpointing_use_reentrant (`bool`, *optional*, defaults to `False`):
+        Whether to apply `use_reentrant` for gradient_checkpointing.
+    ignore_bias_buffers (`bool`, *optional*, defaults to `False`):
+        Debug argument for distributed training. Fix for DDP issues with LM bias/mask buffers - invalid scalar type,
```
Comment (qgallouedec) [inline trl/utils.py:35]: It seems to be a debug argument. Should we remove it like in #2055?

File: trl/utils.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        inplace operation. See https://github.com/huggingface/transformers/issues/22482#issuecomment-1595790992.
+    """
+
+    dataset_name: str
+    dataset_train_split: str = "train"
+    dataset_test_split: str = "test"
+    config: Optional[str] = None
+    gradient_checkpointing_use_reentrant: bool = False
```
Comment (qgallouedec) [inline trl/utils.py:43]: A lot of scripts set it to true. Should True be the default value?

File: docs/source/rloo_trainer.md [hunk 3]
```diff
@@ -18,6 +18,7 @@ To just run a RLOO script to make sure the trainer can run, you can run the foll
 
 ```bash
 python examples/scripts/rloo/rloo.py \
+    --dataset_name trl-lib/chatbot_arena_completions \
```
Comment (qgallouedec) [inline docs/source/rloo_trainer.md:21]:
```suggestion
    --dataset_name trl-internal-testing/descriptiveness-sentiment-trl-style \
```

File: tests/test_rloo_trainer.py [hunk 4]
```diff
@@ -26,6 +26,7 @@
 def test():
     command = """\
 python examples/scripts/rloo/rloo.py \
+    --dataset_name trl-lib/chatbot_arena_completions \
```
Comment (qgallouedec) [inline tests/test_rloo_trainer.py:29]:
```suggestion
    --dataset_name trl-internal-testing/descriptiveness-sentiment-trl-style \
```

File: examples/scripts/rloo/rloo.py [hunk 5]
```diff
@@ -87,7 +88,7 @@
     ################
     # Dataset
     ################
-    dataset = load_dataset("trl-internal-testing/descriptiveness-sentiment-trl-style", split="descriptiveness")
+    dataset = load_dataset(script_args.dataset_name, split="descriptiveness")
```
Comment (edbeeching) [inline examples/scripts/rloo/rloo.py:91]: We could use `split=script_args.train_split` here so the user can run with another dataset?

File: trl/utils.py (hunk 2 above)
Comment (edbeeching) [reply trl/utils.py:43]: You are referring to `gradient_checkpointing_use_reentrant`? I think this is usually set to False so the default is fine.

File: examples/scripts/ppo/ppo.py [hunk 6]
```diff
@@ -86,7 +88,7 @@
     ################
     # Dataset
     ################
-    dataset = load_dataset("trl-internal-testing/descriptiveness-sentiment-trl-style", split="descriptiveness")
+    dataset = load_dataset(script_args.dataset_name, split="descriptiveness")
```
Comment (edbeeching) [inline examples/scripts/ppo/ppo.py:91]: We could use split=script_args.dataset_train_split here so the user can run with another dataset?

**Review by edbeeching — APPROVED** [review]
Thanks for harmonizing all this, LGTM apart from 1 change about the split for some of the RL examples, if you change it be sure to update the docs etc.

File: examples/scripts/rloo/rloo.py (hunk 5 above)
Comment (qgallouedec) [reply examples/scripts/rloo/rloo.py]: Done in 747fb7ca9ae148844b30f18c8a5136f53ce8fac8

File: examples/scripts/ppo/ppo.py (hunk 6 above)
Comment (qgallouedec) [reply examples/scripts/ppo/ppo.py]: Done in 747fb7ca9ae148844b30f18c8a5136f53ce8fac8

**Pull request merged**